app/controllers/CheckYourAnswersController.scala

```diff
@@ -62,13 +65,18 @@ class CheckYourAnswersController @Inject()(appConfig: FrontendAppConfig,
 
   def onSubmit(): Action[AnyContent] = (getData andThen requireData).async {
     implicit request =>
-
       implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-      submissionService.ctutrSubmission(request.userAnswers) map {
-        case SubmissionSuccessful => Redirect(routes.ConfirmationController.onPageLoad())
-        case _ =>                    Redirect(routes.FailedToSubmitController.onPageLoad())
+      companyHouseConnector.validateCRN(Submission(request.userAnswers).companyDetails).flatMap( {
+        case Some(false) => Future.successful(Redirect(routes.CompanyDetailsNoMatchController.onPageLoad()))
+        case Some(true) =>
+          submissionService.ctutrSubmission(request.userAnswers).map {
+            case SubmissionSuccessful => (Redirect(routes.ConfirmationController.onPageLoad()))
+            case _ =>                    (Redirect(routes.FailedToSubmitController.onPageLoad()))
+          }
+        case _ => Future.successful(Redirect(routes.FailedToSubmitController.onPageLoad()))
       }
+      )
```

Comment:
```suggestion
      companyHouseConnector.validateCRN(Submission(request.userAnswers).companyDetails).flatMap {
        case Some(false) => Future.successful(Redirect(routes.CompanyDetailsNoMatchController.onPageLoad()))
        case Some(true) =>
          submissionService.ctutrSubmission(request.userAnswers).map {
            case SubmissionSuccessful => (Redirect(routes.ConfirmationController.onPageLoad()))
            case _ =>                    (Redirect(routes.FailedToSubmitController.onPageLoad()))
          }
        case _ => Future.successful(Redirect(routes.FailedToSubmitController.onPageLoad()))
      }
```
Redundant parentheses here